File: app/src/main/java/com/linksu/android_kotlin_designpatterns/builder_design/Computer.java
package com.linksu.android_kotlin_designpatterns.builder_design;

/**
 * ==========================================
 * 作 者：linksu (sufululove@gmail.com)
 * 版 本：1.0
 * 创建日期： 2017/8/21
 * 描 述：产品的抽象类
 * 修订历史：
 * ==========================================
 */

public abstract class Computer {
    protected String mBoard;
    protected String mDisplay;
    protected String mOS;

    public Computer() {
    }

    //设置主板
    public void setmBoard(String mBoard) {
        this.mBoard = mBoard;
    }

    //设置显示器
    public void setmDisplay(String mDisplay) {
        this.mDisplay = mDisplay;
    }

    //设置操作系统
    public abstract void setOS();

    @Override
    public String toString() {
        return "Computer{" +
                "mBoard='" + mBoard + '\'' +
                ", mDisplay='" + mDisplay + '\'' +
                ", mOS='" + mOS + '\'' +
                '}';
    }
}

File: kotlin_designpatterns/src/main/java/com/linksu/kotlin_designpatterns/builderdesign/Computer.kt
package com.linksu.kotlin_designpatterns.builderdesign

/**
 * Created by linksus on 8/23 0023.
 * 设置
 */
abstract class Computer {
    //设置主板
    data class setBoard(val mboard: String)

    //设置显示器
    data class setDisplay(val mDisplay: String)

    //设置操作系统
    public abstract fun setOS()
}